src/legacy/ui/public/saved_objects/helpers/save_saved_object.ts

```diff
@@ -0,0 +1,123 @@
+function isErrorNonFatal(error: { message: string }) {
+  if (!error) return false;
+  return error.message === OVERWRITE_REJECTED || error.message === SAVE_DUPLICATE_REJECTED;
+}
+
+/**
+ * Saves this object.
+ *
+ * @param {string} [esType]
+ * @param {SavedObject} [savedObject]
+ * @param {SavedObjectsClient} [savedObjectsClient]
+ * @param {SavedObjectConfig} [config]
+ * @param {object} [options={}]
+ * @property {boolean} [options.confirmOverwrite=false] - If true, attempts to create the source so it
+ * can confirm an overwrite if a document with the id already exists.
+ * @property {boolean} [options.isTitleDuplicateConfirmed=false] - If true, save allowed with duplicate title
+ * @property {func} [options.onTitleDuplicate] - function called if duplicate title exists.
+ * When not provided, confirm modal will be displayed asking user to confirm or cancel save.
+ * @return {Promise}
+ * @resolved {String} - The id of the doc
+ */
+export async function saveSavedObject(
+  savedObject: SavedObject,
+  savedObjectsClient: SavedObjectsClientContract,
+  config: SavedObjectConfig,
+  {
+    confirmOverwrite = false,
+    isTitleDuplicateConfirmed = false,
+    onTitleDuplicate,
+  }: SavedObjectSaveOpts = {}
+): Promise<string> {
+  const esType = config.type || '';
+  const extractReferences = config.extractReferences;
+  // Save the original id in case the save fails.
+  const originalId = savedObject.id;
+  // Read https://github.com/elastic/kibana/issues/9056 and
+  // https://github.com/elastic/kibana/issues/9012 for some background into why this copyOnSave variable
+  // exists.
+  // The goal is to move towards a better rename flow, but since our users have been conditioned
+  // to expect a 'save as' flow during a rename, we are keeping the logic the same until a better
+  // UI/UX can be worked out.
+  if (savedObject.copyOnSave) {
+    savedObject.id = null;
+  }
+
+  // Here we want to extract references and set them within "references" attribute
+  let { attributes, references } = savedObject._serialize();
+  if (extractReferences) {
+    ({ attributes, references } = extractReferences({ attributes, references }));
+  }
+  if (!references) throw new Error('References not returned from extractReferences');
+
+  try {
+    await checkForDuplicateTitle(
+      savedObject,
+      savedObjectsClient,
+      isTitleDuplicateConfirmed,
+      onTitleDuplicate
+    );
+    savedObject.isSaving = true;
+    const resp = confirmOverwrite
+      ? await createSource(
+          attributes,
+          savedObject,
+          savedObjectsClient,
+          esType,
+          savedObject.creationOpts({ references })
+        )
+      : await savedObjectsClient.create(
+          esType,
+          attributes,
+          savedObject.creationOpts({ references, overwrite: true })
+        );
+
+    savedObject.id = resp.id;
+    if (savedObject.showInRecentlyAccessed && savedObject.getFullPath) {
+      npStart.core.chrome.recentlyAccessed.add(
```

Comment:
Can we pass in chrome as a parameter here?

Reply:
definitely! thx! done!